src/app/editor/shared/services/record-fixer.service.ts

```diff
@@ -27,46 +27,81 @@ import { JsonUtilService } from './json-util.service'
 
 @Injectable()
 export class RecordFixerService {
-  
+
   private emptyValueService: EmptyValueService = new EmptyValueService();
   private jsonUtilService: JsonUtilService = new JsonUtilService();
 
+  // TODO: return fixed record!
   fixRecord(record: Object, schema: Object) {
-    Object.keys(schema).forEach(field => this.fix(field, record, schema[field]));
+    Object.keys(record).forEach(field => {
+      if (!schema[field]) {
+        this.deleteField(record, field);
+      } else {
+        this.fix(field, record, schema[field])
+      }
+    });
+
+    this.insertEmptyIntoAlwaysShowFields(record, schema);
+  }
+
+
+  private insertEmptyIntoAlwaysShowFields(record: Object, schema: Object) {
+    let paths = new AlywaysShowPathFinder().getPaths(schema);
+    paths.forEach(path => {
+      let rootSchema = { 'properties': schema }; // HACK!
+      this.insertEmptyIntoPath(path, record, rootSchema);
+    });
   }
 
+  private insertEmptyIntoPath(path: Array<string>, value: any, schema: Object) {
```

Comment:
Might be enough just to add a docstring to the function explaining what it does and documenting what each parameter is.